### PR TITLE
Document the choices of OTA operations

### PR DESCRIPTION
**Pull request by Tim-Anderson**
## Readiness

* [x] Merge (pending reviews)
* [ ] Merge after _date or event_
* [x ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [ ] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 9e0a9a8c1761cb9a6ecb15c75303df24e0ec11bb are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2576/docs/artifacts/html/index.html

File: source/user-guide/custom-sota-client.rst [hunk 1]
```diff
@@ -1,7 +1,51 @@
...
+================================
+
+The Secure Over The Air update (SOTA), by default operates as a daemon process (ref:`ref-aktualizr-lite`) which
+periodically checks for updates. If an update is available it will automatically download, and install
+it to a device that is following the update tag.
+
+This is not always the desired operation. There are 3 ways to control this operation
+
+1. Callbacks
+2. aktualizr-lite manual command line interface (Client)
```
Comment (vanmaegima) [inline source/user-guide/custom-sota-client.rst:13]: I think we decided not to advertise that for now?
Maybe we can drop this and focus on custom sota x callbacks and check our customer feedback.

File: source/user-guide/custom-sota-client.rst [hunk 2]
```diff
@@ -1,7 +1,51 @@
...
+   https://github.com/foundriesio/meta-lmp//blob/main/recipes-sota/aktualizr/aktualizr-callback/callback-handler
+
+Aktualizr-lite CLI
+------------------
+
+Operating aktualizr-lite in `manual mode`_. This allows calling aktualizr-lite with specific commands to
+step through the install process. Unlike the callback method, this uses return codes to provice the status and
+does not perform the reboot of the device automatically.
+
+.. _`manual mode`:
```
Comment (vanmaegima) [inline source/user-guide/custom-sota-client.rst:44]: Same comment here, we were always against running manual commands as we cannot test every single interaction (we also changed our docs a while ago stating this is not supported and not recommended).

**Comment by vanmaegima** [comment]
Thank you @Tim-Anderson for the help with this. I would say to remove CLI stuff until we have a chance to discuss with Mike/Mike Sul. Callbacks docs is what we are lacking at the moment.

**Comment by doanac** [comment]
Docs for ae95aac86d0a44ffdf34d9f8353f95fc4aa9e658 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2580/docs/artifacts/html/index.html

File: source/user-guide/custom-sota-client.rst [hunk 3]
```diff
@@ -1,7 +1,40 @@
...
+
+#. Callbacks
+#. Custom Update Agent
+
+Callbacks
+---------
+
+The aktualizr-lite provides the ability to run an executable at the following OTA operations:
+
+* Before checking in - check-for-update-pre  return: none
```
Comment (kprosise) [inline source/user-guide/custom-sota-client.rst:20]:
```suggestion
* Before checking in — check-for-update-pre  return: none
```

File: source/user-guide/custom-sota-client.rst [hunk 4]
```diff
@@ -1,7 +1,40 @@
...
+* After checking in  - check-for-update-post return: OK or FAILED: reason
+* Before a download  - download-pre          return: none
+* After a download   - download-post         return: OK or FAILED: reason
+* Before an install  - install-pre           return: none
+* After an install   - install-post          return: NEEDS_COMPLETION, OK, or FAILED: reason
+* After a reboot     - install-final-pre     return: none
```
Comment (kprosise) [inline source/user-guide/custom-sota-client.rst:26]:
```suggestion
* After checking in  — check-for-update-post return: OK or FAILED: reason
* Before a download  — download-pre          return: none
* After a download   — download-post         return: OK or FAILED: reason
* Before an install  — install-pre           return: none
* After an install   — install-post          return: NEEDS_COMPLETION, OK, or FAILED: reason
* After a reboot     — install-final-pre     return: none
```

**Review by kprosise — APPROVED** [review]
Style/grammar suggestions, otherwise it lgtm

**Comment by vanmaegima** [comment]
@Tim-Anderson What are we missing on this?
@kprosise @foundriesio/customer-success Do you think it's a good idea to land this on v93?

**Comment by kprosise** [comment]
@vanmaegima I have no problem including this in v93 if it is ready then.

**Comment by doanac** [comment]
Docs for 91074ce140ef493a1e0643f02b62ccf511aeba47 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2624/docs/artifacts/html/index.html

**Review by vanmaegima — APPROVED** [review]

**Review by MrCry0 — APPROVED** [review]

**Comment by kprosise** [comment]
LGTM

**Pull request merged**